File: AppDev/Java/Jokes-App/app/src/main/java/com/example/jokesapitester/MainActivity.java
package com.example.jokesapitester;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import org.json.JSONObject;

import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class MainActivity extends AppCompatActivity {
    public class DownloadTask extends AsyncTask<String, Void, String> {

        @Override
        protected String doInBackground(String... urls) {
            StringBuilder result = new StringBuilder();
            URL url;
            HttpURLConnection urlConnection;

            try {
                url = new URL(urls[0]);
                urlConnection = (HttpURLConnection) url.openConnection();
                InputStream in = urlConnection.getInputStream();
                InputStreamReader reader = new InputStreamReader(in);
                int data = reader.read();
                while (data != -1) {
                    char current = (char) data;
                    result.append(current);
                    data = reader.read();
                }
                return result.toString();
            } catch (Exception e) {
                e.printStackTrace();
                return "failed";
            }

        }
        protected void onPostExecute(String s) {
            super.onPostExecute(s);
            EditText setup=findViewById(R.id.text1);
            EditText punchline=findViewById(R.id.texto2);
            try {
                JSONObject jsonObject=new JSONObject(s);
                String message=jsonObject.getString("setup");
                setup.setText(message);
                String message2=jsonObject.getString("punchline");
                punchline.setText(message2);




            } catch (Exception e) {
                e.printStackTrace();

                Toast.makeText(getApplicationContext(),"could not get api", Toast.LENGTH_SHORT).show();
            }
        }


    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


    }

    public void press(View view){
        DownloadTask task = new DownloadTask();
        //task.execute("https://official-joke-api.appspot.com/random_joke");
        task.execute("https://nova-joke-api.netlify.app/.netlify/functions/index/random_joke");
        InputMethodManager mgr= (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
    }

}

File: AppDev/Kotlin/User friendly Calculator/app/src/main/java/com/example/calculator/MainActivity.kt
package com.example.calculator

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.ArithmeticException

class MainActivity : AppCompatActivity() {
    var lastnumeric= false
    var lastdot= false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    fun onDigit(view: View){
        TvInput.append((view as Button).text)
        lastnumeric=true
    }
    fun onclear(view :View){
        TvInput.text=""
        lastnumeric=false
        lastdot=false
    }
    fun ondot(view: View){
        if(lastnumeric && !lastdot) {
            TvInput.append((view as Button).text)
            lastnumeric=false
            lastdot=true
        }
    }

    fun onoperator(view:View)
    {
        if(lastnumeric && !isoperatoradded(TvInput.text.toString())){
            TvInput.append((view as Button).text)
            lastnumeric=false
            lastdot=false
        }
    }
    fun removezero(result: String):String{
        var value= result
        if(value.contains(".0"))
            value=result.substring(0,result.length-2)
        return value

    }
    private fun isoperatoradded(value : String):Boolean{
        return if (value.startsWith("-"))
        {false}
        else{
            value.contains("/") || value.contains("*") || value.contains("+") || value.contains("-")
        }
    }
    fun onequal(view: View){
        if(lastnumeric){
            var tvvalue = TvInput.text.toString()
            var prefix =""
            if(tvvalue.startsWith("-")){
                prefix="-"
                tvvalue=tvvalue.substring(1)
            }
            try{
                if(tvvalue.contains("-")){
                    val splitval = tvvalue.split("-")
                    var one = splitval[0]
                    var two = splitval[1]
                    if(!prefix.isEmpty()){
                        one=prefix+one
                    }
                    TvInput.text = removezero((one.toDouble()-two.toDouble()).toString())
                 }
                else if(tvvalue.contains("+")){
                    val splitval = tvvalue.split("+")
                    var one = splitval[0]
                    var two = splitval[1]
                    if(!prefix.isEmpty()){
                        one=prefix+one
                    }
                    TvInput.text = removezero((one.toDouble()+two.toDouble()).toString())
                }
                else if(tvvalue.contains("*")){
                    val splitval = tvvalue.split("*")
                    var one = splitval[0]
                    var two = splitval[1]
                    if(!prefix.isEmpty()){
                        one=prefix+one
                    }
                    TvInput.text = removezero((one.toDouble()*two.toDouble()).toString())
                }
                else if(tvvalue.contains("/")){
                    val splitval = tvvalue.split("/")
                    var one = splitval[0]
                    var two = splitval[1]
                    if(!prefix.isEmpty()){
                        one=prefix+one
                    }
                    TvInput.text = removezero((one.toDouble()/two.toDouble()).toString())
                }
            }catch (e: ArithmeticException){
                e.printStackTrace()
            }
        }
    }

}